### PR TITLE
Add issue and PR templates, document RFC process

### PR DESCRIPTION
**Pull request by hdevalence**
Also fixes the numbering for the async script RFC.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE/pull_request.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+you're trying to solve? In some cases there is not a problem and this can be
+thought of as being the motivation for your change.
+-->
+
+## Solution
+
+<!--
+Summarize the solution and provide any necessary context needed to understand
+the code change.
+-->
```
Comment (dconnolly) [inline .github/PULL_REQUEST_TEMPLATE/pull_request.md:23]:
```suggestion
-->

## Related Issues
<!--
Please link to any existing GitHub issues pertaining to this PR.
-->
```

File: .github/PULL_REQUEST_TEMPLATE/rfc.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+assignees: ''
+
+---
+
+<!--
+This template is for design RFCs.  The RFC process is documented in CONTRIBUTING.md
+
+- [ ] Copy the fields from the text of your RFC below
+- [ ] Leave the Design PR number blank, then fill it in *after* you create the PR (and the issue number is created)
+- [ ] After creating the Design PR, take the next available RFC number (not conflicting with any lower-number RFCs) and name your RFC book/rfcs/XYZT-my-rfc.md).
```
Comment (dconnolly) [inline .github/PULL_REQUEST_TEMPLATE/rfc.md]: Name the PR name? Or the filename of the .md ?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,18 +1,65 @@
...
 ## Pull Requests
 [pull-requests]: #pull-requests
 
-## Writing Documentation
-[writing-documentation]: #writing-documentation
+PRs are welcome for small and large changes, but please don't make large PRs
+without coordinating with us via the issue tracker or Discord. This helps
+increase development coordination and makes PRs easier to merge.
+
+Check out the [help wanted][hw] or [good first issue][gfi] labels if you're
```
Comment (dconnolly) [inline CONTRIBUTING.md:27]: 💯

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,18 +1,65 @@
...
+6. Take the next available RFC number (not conflicting with any existing RFCs
+or design PRs) and name the RFC file accordingly, e.g., `0027-my-feature.md`
+for number 27. Make sure that `book/src/SUMMARY.md` links to the numbered RFC.
+
+7. After creating an RFC PR, update the RFC header and the PR description
+with the PR number.
+
+8. After the RFC is accepted, create an issue for the implementation of the
+design, and update the RFC header and PR description with the implementation
+issue number.
```
Comment (dconnolly) [inline CONTRIBUTING.md:64]: This duplicates some of the instructions in the PR template itself; is there a way to write it once and reference elsewhere, to keep them in sync?

Comment (hdevalence) [reply CONTRIBUTING.md:64]: I'm not sure of a way to keep these in sync, we could refer from one to the other but I thought that a little duplication could be good in order to have a checklist right there in the template.  No strong feelings etither way.

File: .github/PULL_REQUEST_TEMPLATE/rfc.md (hunk 2 above)
Comment (hdevalence) [reply .github/PULL_REQUEST_TEMPLATE/rfc.md]: Should be the filename of the .md, I'll clarify.  Actually this was correctly described in the Contributing file so I think your comment below is a good concern.  Maybe I should edit this to just refer to that description?

Comment (hdevalence) [reply .github/PULL_REQUEST_TEMPLATE/rfc.md]: Edited to just refer to the description.

File: CONTRIBUTING.md (hunk 4 above)
Comment (hdevalence) [reply CONTRIBUTING.md:64]: Deduplicated by referring to CONTRIBUTING

File: .github/PULL_REQUEST_TEMPLATE/rfc.md [hunk 5]
```diff
@@ -8,12 +8,11 @@ assignees: ''
...
-This template is for design RFCs.  The RFC process is documented in CONTRIBUTING.md
+This template is for design RFCs.
 
-- [ ] Copy the fields from the text of your RFC below
-- [ ] Leave the Design PR number blank, then fill it in *after* you create the PR (and the issue number is created)
-- [ ] After creating the Design PR, take the next available RFC number (not conflicting with any lower-number RFCs) and name your RFC book/rfcs/XYZT-my-rfc.md).
-- [ ] Leave the Zebra Issue number blank.  *After* the RFC is merged, create an issue to track implementation of the RFC and edit the PR description for cross-referencing.
+The RFC process is documented in CONTRIBUTING.md.
+
+Please see the checklist there: https://zebra.zfnd.org/CONTRIBUTING.html
```
Comment (dconnolly) [inline .github/PULL_REQUEST_TEMPLATE/rfc.md:15]: 👍

**Review by dconnolly — APPROVED** [review]

**Pull request merged**